pkg/storage/localfs/store.go

```diff
@@ -155,9 +158,70 @@ func (l *localFS) Keys(ctx context.Context) ([]string, error) {
 	return res, nil
 }
 
-//TODO discuss the implementation with @Ivan & @Ritesh
-func (l *localFS) KeysPrefix(ctx context.Context, token, prefix, delimiter string, count int) ([]string, string, error) {
-	return nil, "", storagestatus.ErrNotImplemented
+// KeyPrefix provides a paginated key iterator using "pageToken" as the next starting point
+//
+// NOTE: "delimiter" is ignored (always set to "/").
+//
+// TODO: This implementation does not really scale up, but it is quite workable for our testcases using locafs.
+func (l *localFS) KeysPrefix(_ context.Context, token, prefix, _ string, count int) ([]string, string, error) {
+	l.exclusive.Lock()
+	defer l.exclusive.Unlock()
+
+	prefix = path.Clean("/" + prefix)
```

Comment:
Afero api reminders:

* even in case of `BasePathFS`, `afero.Walk` can accept an absolute path as `root`
* when `Walk` is provided an absolute path, the `WalkFunc`'s `pth` parameter is also an absolute path

Reply:
yes I've tried to normalize things here: input always goes absolute, output always go relative...